components/brave_shields/browser/domain_block_controller_client.cc

```diff
@@ -0,0 +1,69 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_shields/browser/domain_block_controller_client.h"
+
+#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
+#include "brave/components/brave_shields/browser/domain_block_tab_storage.h"
+#include "components/prefs/pref_service.h"
+#include "components/security_interstitials/content/settings_page_helper.h"
+#include "components/security_interstitials/core/metrics_helper.h"
+#include "content/public/browser/page_navigator.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/common/referrer.h"
+#include "ui/base/page_transition_types.h"
+#include "ui/base/window_open_disposition.h"
+
+namespace brave_shields {
+
+// static
+std::unique_ptr<security_interstitials::MetricsHelper>
+DomainBlockControllerClient::GetMetricsHelper(const GURL& url) {
+  security_interstitials::MetricsHelper::ReportDetails report_details;
+  report_details.metric_prefix = "domain_block";
+
+  return std::make_unique<security_interstitials::MetricsHelper>(
+      url, report_details, nullptr);
+}
+
+DomainBlockControllerClient::DomainBlockControllerClient(
+    content::WebContents* web_contents,
+    const GURL& request_url,
+    AdBlockCustomFiltersService* ad_block_custom_filters_service,
+    PrefService* prefs,
+    const std::string& locale)
+    : security_interstitials::SecurityInterstitialControllerClient(
+          web_contents,
+          GetMetricsHelper(request_url),
+          prefs,
+          locale,
+          GURL("about:blank") /* default_safe_page */,
+          nullptr /* settings_page_helper */),
+      request_url_(request_url),
+      ad_block_custom_filters_service_(ad_block_custom_filters_service),
+      dont_warn_again_(false) {}
+
+void DomainBlockControllerClient::GoBack() {
+  SecurityInterstitialControllerClient::GoBackAfterNavigationCommitted();
+}
+
+void DomainBlockControllerClient::Proceed() {
+  DomainBlockTabStorage* tab_storage =
+      DomainBlockTabStorage::GetOrCreate(web_contents_);
+  tab_storage->SetIsProceeding(true);
+  if (dont_warn_again_) {
+    std::string custom_filters =
+        ad_block_custom_filters_service_->GetCustomFilters();
+    ad_block_custom_filters_service_->UpdateCustomFilters(
+        "@@||" + request_url_.host() + "^\n" + custom_filters);
```

Comment:
Random idea I had, but could we potentially add some kind of comment along with any auto-generated filters? e.g.
```
! Suspicious site exception added 2021-02-19T23:20:32Z
@@||example.com^
```

I feel this would especially be good once we move the `Right-click > Brave > Block element via selector` blocking to use the custom filters box as well.

Reply:
cc @diracdeltas - I wouldn't anticipate it being problematic, but arguably this could be construed as a way of storing browsing history over time?

Reply:
Ooh, I like that idea very much.

Reply:
Is that exactly what it should say?

Reply:
Personally, that's what I would make it say :smile:

Reply:
This data is only stored in response to direct user action (clicking "don't warn again about this site" on the interstitial), and the feature can not be implemented without storing the domain. The question here is whether we should also add a date/timestamp with each domain the user chooses to block.

Reply:
creating a followup issue to track this

Reply:
@pilgrim-brave could you link the issue here? I don't see it